sqlglot/optimizer/remove_unused_columns.py

```diff
@@ -0,0 +1,162 @@
+from collections import defaultdict
```

Comment:
let's call this projection_pushdown

Reply:
👍 
I had it that at first, then changed it to match https://github.com/duckdb/duckdb/blob/master/src/optimizer/remove_unused_columns.cpp
I'll switch it back